cmd/vic-machine/common/proxy.go

```diff
@@ -52,19 +52,25 @@ func (p *Proxies) ProcessProxies() (hproxy, sproxy *url.URL, err error) {
 		p.IsSet = true
 	}
 	if p.HTTPProxy != nil && *p.HTTPProxy != "" {
-		hproxy, err = url.Parse(*p.HTTPProxy)
-		if err != nil || hproxy.Host == "" || hproxy.Scheme != "http" {
-			err = cli.NewExitError(fmt.Sprintf("Could not parse HTTP proxy - expected format http://fqnd_or_ip:port: %s", *p.HTTPProxy), 1)
+		hproxy, err = p.validate(*p.HTTPProxy)
+		if err != nil {
 			return
 		}
 	}
 
 	if p.HTTPSProxy != nil && *p.HTTPSProxy != "" {
-		sproxy, err = url.Parse(*p.HTTPSProxy)
-		if err != nil || sproxy.Host == "" || sproxy.Scheme != "https" {
-			err = cli.NewExitError(fmt.Sprintf("Could not parse HTTPS proxy - expected format https://fqnd_or_ip:port: %s", *p.HTTPSProxy), 1)
-			return
-		}
+		sproxy, err = p.validate(*p.HTTPSProxy)
+	}
+	return
+}
+
+func (p *Proxies) validate(ref string) (proxy *url.URL, err error) {
```

Comment:
Can we have a unit test that ensures the validation is successful for the various cases?
We do not have a test for omitting the scheme currently and I recall there was a `url.Parse` problem if scheme was omitted - the host portion could end up as the scheme.

I suggest testing at a minimum:
```
https://fully.qualified.domain.name
https://fully.qualified.domain.name:443
http://fully.qualified.domain.name
http://fully.qualified.domain.name:80
fully.qualified.domain.name
fully.qualified.domain.name:80
raw-host
raw-host:80
```

Reply:
Good point. I will add one.